File: forge/src/main/java/juuxel/adorn/platform/forge/BlockEntityBridgeForge.java
package juuxel.adorn.platform.forge;

import juuxel.adorn.block.entity.BrewerBlockEntity;
import juuxel.adorn.block.entity.KitchenSinkBlockEntity;
import juuxel.adorn.platform.BlockEntityBridge;
import juuxel.adorn.platform.forge.block.entity.BrewerBlockEntityForge;
import juuxel.adorn.platform.forge.block.entity.KitchenSinkBlockEntityForge;
import net.minecraft.world.level.block.state.BlockState;
import net.minecraft.core.BlockPos;

public final class BlockEntityBridgeForge implements BlockEntityBridge {
    public static final BlockEntityBridgeForge INSTANCE = new BlockEntityBridgeForge();

    @Override
    public BrewerBlockEntity createBrewer(BlockPos pos, BlockState state) {
        return new BrewerBlockEntityForge(pos, state);
    }

    @Override
    public KitchenSinkBlockEntity createKitchenSink(BlockPos pos, BlockState state) {
        return new KitchenSinkBlockEntityForge(pos, state);
    }
}

File: forge/src/main/kotlin/juuxel/adorn/platform/forge/BlockEntityBridgeForge.kt
package juuxel.adorn.platform.forge

import juuxel.adorn.block.entity.BrewerBlockEntity
import juuxel.adorn.block.entity.KitchenSinkBlockEntity
import juuxel.adorn.platform.BlockEntityBridge
import juuxel.adorn.platform.forge.block.entity.BrewerBlockEntityForge
import juuxel.adorn.platform.forge.block.entity.KitchenSinkBlockEntityForge
import net.minecraft.block.BlockState
import net.minecraft.util.math.BlockPos

object BlockEntityBridgeForge : BlockEntityBridge {
    override fun createBrewer(pos: BlockPos, state: BlockState): BrewerBlockEntity =
        BrewerBlockEntityForge(pos, state)

    override fun createKitchenSink(pos: BlockPos, state: BlockState): KitchenSinkBlockEntity =
        KitchenSinkBlockEntityForge(pos, state)
}
